src/src_cryptography/src_kmc_crypto_service/cryptography_interface_kmc_crypto_service.template.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2021, by the California Institute of Technology.
+ * ALL RIGHTS RESERVED. United States Government Sponsorship acknowledged.
+ * Any commercial use must be negotiated with the Office of Technology
+ * Transfer at the California Institute of Technology.
+ *
+ * This software may be subject to U.S. export control laws. By accepting
+ * this software, the user agrees to comply with all applicable U.S.
+ * export laws and regulations. User has the responsibility to obtain
+ * export licenses, or other export authority as may be required before
+ * exporting such information to foreign countries or providing access to
+ * foreign persons.
+ */
+
+#include "crypto_error.h"
+#include "cryptography_interface.h"
+
+#include <stdio.h>
+
+// Cryptography Interface Initialization & Management Functions
+static int32_t cryptography_config(void);
+static int32_t cryptography_init(void);
+static crypto_key_t* get_ek_ring(void);
+static int32_t cryptography_shutdown(void);
+// Cryptography Interface Functions
+static int32_t cryptography_encrypt(void);
+static int32_t cryptography_decrypt(void);
+static int32_t cryptography_authenticate(void);
+static int32_t cryptography_validate_authentication(void);
+static int32_t cryptography_aead_encrypt(uint8_t* data_out, uint32_t len_data_out,
+                                         uint8_t* data_in, uint32_t len_data_in,
+                                         uint8_t* key, uint32_t len_key,
+                                         SecurityAssociation_t* sa_ptr,
+                                         uint8_t* iv, uint32_t iv_len,
+                                         uint8_t* mac, uint32_t mac_size,
+                                         uint8_t* aad, uint32_t aad_len,
+                                         uint8_t encrypt_bool, uint8_t authenticate_bool,
+                                         uint8_t aad_bool);
+static int32_t cryptography_aead_decrypt(uint8_t* data_out, uint32_t len_data_out,
+                                         uint8_t* data_in, uint32_t len_data_in,
+                                         uint8_t* key, uint32_t len_key,
+                                         SecurityAssociation_t* sa_ptr,
+                                         uint8_t* iv, uint32_t iv_len,
+                                         uint8_t* mac, uint32_t mac_size,
+                                         uint8_t* aad, uint32_t aad_len,
+                                         uint8_t decrypt_bool, uint8_t authenticate_bool,
+                                         uint8_t aad_bool);
+/*
+** Global Variables
+*/
+// Cryptography Interface
+static CryptographyInterfaceStruct cryptography_if_struct;
```

Comment:
Should these globals be here, or should they be declared in the crypto_interface.c ? --Per J.Lucas

Reply:
It's not a global actually, it's a static (local/module) variable... Sorry, I didn't update the comment here, but I did in the libgcrypt interface file:
https://github.com/nasa/CryptoLib/pull/46/files#diff-8455637f5a51288fb98808edae34ef7ba4faf362e80d728810678ec2701f1399R51

This is the local struct that contains all of the function implementations for the cryptography interface. It is defined with different function pointers depending on which interface is being used, and returned to the core application. This should be here and is really central to how the interface layer is implemented, IE, via these locally defined structs with different function pointers.